infinitewisdom/analysis.py

```diff
@@ -40,19 +54,28 @@ class GoogleVision(ImageAnalyser):
     Google Vision API implementation
     """
 
-    def __init__(self, auth_file: str):
+    def __init__(self, auth_file_path: str):
         """
-        :param auth_file: authentication file for the google vision api
+        :param auth_file_path: authentication file for the google vision api
         """
         # Imports the Google Cloud client library
         from google.cloud import vision
 
+        self._auth_file_path = auth_file_path
+        self._auth_id = os.path.split(auth_file_path)[1]
+
         # Instantiates a client
-        self._client = vision.ImageAnnotatorClient.from_service_account_file(auth_file)
+        self._client = vision.ImageAnnotatorClient.from_service_account_file(auth_file_path)
 
     def get_identifier(self) -> str:
         return "google-vision"
 
+    def get_quality(self) -> float:
+        return 0.9
+
+    def get_monthly_capacity(self) -> float:
+        return 1000.0
```

Comment:
It should be possible to override this in the configuration file. A few use cases:

1. I run two instances of this bot and during development I may start an additional instance. I would like to configure different limits for those instances, e.g. production: 800, stage: 100 and development: 100. That way I would stay below the global limit of thousand queries.
2. In case I loose the persisted data (hopefully I won't) the bot will not be able to count the previous API queries. In this case I could manually set a lower limit to avoid additional costs.
3. I am willing to pay a few bucks to analyse more images so in this case it would be required to raise the limit.

Reply:
> It should be possible to override this in the configuration file. A few use cases:

Sure. I can add a config parameter for this. 

> 1. I run two instances of this bot and during development I may start an additional instance. I would like to configure different limits for those instances, e.g. production: 800, stage: 100 and development: 100. That way I would stay below the global limit of thousand queries.

If I understand you correctly this only applies to multiple instances using different persistence instances.

> 2. In case I loose the persisted data (hopefully I won't) the bot will not be able to count the previous API queries. In this case I could manually set a lower limit to avoid additional costs.

This could also be prevented if it is possible to query the API how many requests were made so far. Such a method would not fit the current abstraction of `ImageAnalyser` though so it might be preferable to investigate such a solution to reduce refactoring.

Reply:
> > It should be possible to override this in the configuration file. A few use cases:
> 
> Sure. I can add a config parameter for this.

Thanks :)

> > 1. I run two instances of this bot and during development I may start an additional instance. I would like to configure different limits for those instances, e.g. production: 800, stage: 100 and development: 100. That way I would stay below the global limit of thousand queries.
> 
> If I understand you correctly this only applies to multiple instances using different persistence instances.

Yes, exactly.

> > 1. In case I loose the persisted data (hopefully I won't) the bot will not be able to count the previous API queries. In this case I could manually set a lower limit to avoid additional costs.
> 
> This could also be prevented if it is possible to query the API how many requests were made so far. Such a method would not fit the current abstraction of `ImageAnalyser` though so it might be preferable to investigate such a solution to reduce refactoring.

Agreed.

As discussed offline I will merge the current state and we will implement this later.